linter.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+COLOR_REST='\e[0m';
+COLOR_BLUE='\e[1;34m';
+
+DIFF_EXE=""
+CLANG_FORMATTER=""
+AUTO_FIX="false"
+
+if [ "${1}" == '-i' ]; then
+	AUTO_FIX="true"
+fi
+
+if command -v colordiff > /dev/null; then
+	DIFF_EXE=$(command -v colordiff)
+elif command -v diff > /dev/null; then
+	DIFF_EXE=$(command -v diff)
+fi
+
+if command -v clang-format > /dev/null; then
+	CLANG_FORMATTER=$(command -v clang-format)
+else
+	echo -e "Missing clang-format, please install it with: \n    ${COLOR_BLUE}sudo apt-get install clang-format${COLOR_REST}"
+	exit 1
+fi
+
+file_list=$(find ini \( -name '*.h' -or -name '*.cpp' \))
```

Comment:
Do you not want to include the tests as well?

Reply:
let's include them